apps/tng-dictionary/src/components/TermIndex/TermIndex.tsx

```diff
@@ -19,155 +18,194 @@ import TermData from '../Term/Term';
-    <TextField
-      value={componentState.searchText}
-      placeholder={placeholderText}
-      onChange={(event) => handleSearchTextChange(event)}
-      InputProps={{
-        className: 'searchProps',
-        endAdornment: (
-          <SearchIcon className='searchIcon' />
-        )
-      }} />)
-
-  // SWITCHCOMPONENT
-
-  const handleSwitchStateChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const newSearchContext = mapSwitchStateToSearchContext(event.target.checked)
-
-    setComponentState({
-      ...componentState,
-      searchContext: newSearchContext,
-      selectedTerms: componentState.searchText ? determineSelectedTerms(componentState.allTerms, { [newSearchContext]: componentState.searchText }) : componentState.allTerms
-    })
-  }
-
-
-  return (
-
-    <ThemeProvider theme={theme}>
-      <motion.div initial={{ opacity: 0 }} animate={{ opacity: 1 }} exit={{ opacity: 0 }} transition={{ duration: .6 }}>
-        <div className='termIndex'>
-          <section className='section'>
-            <h1 className='header'>
-              Terms <MenuBookTwoToneIcon className='headerIcon' />
-            </h1>
-            <div>
-              {buildSearchElement(searchContextToPlaceholder[componentState.searchContext])}
-            </div>
-
-            <FormControlLabel control={<Switch checked={mapSearchContextToSwitchState(componentState.searchContext)} onChange={e => { handleSwitchStateChange(e) }} inputProps={{ 'aria-label': 'controlled' }} />} label={"Tŝilhqot'in / English"} />
-          </section>
-
-          <DataGrid
-            className='grid'
-            rows={rows}
-            columns={columns}
-            rowsPerPageOptions={[10, 50, 100]}
-            initialState={{
-              pagination: {
-                pageSize: 10,
-              }
-            }}
-
-            components={{
-              NoRowsOverlay: () => (
-                <MiniLoading />
-              ),
-              Panel: () => (
-                <p>© 2022 Tŝilhqot’in National Government</p>
-              )
+    const [componentState, setComponentState] = useState<ComponentState>({
+        allTerms: [],
+        searchContext: 'term',
+        searchText: '',
+        selectedTerms: [],
+    });
+
+    useEffect(() => {
+        setComponentState({
+            allTerms: [],
+            searchContext: 'term',
+            searchText: '',
+            selectedTerms: [],
+        });
+        const apiUrl = `http://104.225.142.106:3131/api/entities/terms`;
+        fetch(apiUrl, { mode: 'cors' })
+            .then((res) => res.json())
+            .then((allTerms) => {
+                setComponentState({
+                    ...componentState,
+                    allTerms: allTerms,
+                    selectedTerms: allTerms,
+                });
+            })
+            .catch((rej) => console.log(rej));
+    }, [setComponentState]);
+
+    if (componentState.allTerms === []) return <Loading />;
```

Comment:
We need to have a different behaviour when the search succeeds but finds no results (i.e. returns []). Make a note to do this very soon.